### PR TITLE
SIWPA - 29, 30, 35 Add Seen/Unseen UI for messages, Individual Contact Message Page, Seen / Unseen Message API Route

### PR DESCRIPTION
**Pull request by megfinnegan**
- view full contact message
- seen/unseen for contact inbox messages
- check api routes for seen_message and get_message in admin.py

### CODE REVIEW

File: api/admin.py [hunk 1]
```diff
@@ -175,3 +175,49 @@ def get_messages(message_filter: str = 'all'):
         code = OK
 
     return response, code
+
+# Route to get a singular message with given id.
+@admin.route('/get-message/<message_id>', methods=['GET'])
+def get_message(message_id: int):
+    """Get a singular message with the given id
+    }
```
Comment (jventura1738) [inline api/admin.py:196]: I am assuming you know the form of the message being returned, but if you could add the form here in the comments just for documentation that would be great

Comment (jventura1738) [reply api/admin.py:196]: See under the 'get listing' routes to see what I mean; JSON format

File: api/admin.py [hunk 2]
```diff
@@ -175,3 +175,49 @@ def get_messages(message_filter: str = 'all'):
...
+def seen_message(message_id: int):
+    """Mark a message as seen
+
+    If a message has been seen, remove dot. 
+    """
+    response = dict()
+
+    # Check if listing is in database, then update and return to Jake:
+    if message := ContactFormMessage.query.filter_by(id=message_id).first():
+        message.was_seen = True if message.was_seen is False else False
```
Comment (jventura1738) [inline api/admin.py]: Looks like we have a copy paste error in the comment, and in the logic.  Unless I am wrong, do you want a seen message to be marked as unseen after hitting this route again?

**Review by jventura1738 — CHANGES_REQUESTED** [review]
See comments made, only one of them is major.

File: api/admin.py (hunk 1 above)
Comment (jacobduncan00) [reply api/admin.py:196]: I agree

File: api/admin.py (hunk 2 above)
Comment (jacobduncan00) [reply api/admin.py]: Good point. We don't want to just flip the boolean, we only want to change from True to False and never change again

Comment (megfinnegan) [reply api/admin.py]: I originally had it just change and then must have changed it back at some point on accident. I'll fix it now.

File: api/admin.py [hunk 3]
```diff
@@ -207,9 +220,9 @@ def seen_message(message_id: int):
     """
     response = dict()
 
-    # Check if listing is in database, then update and return to Jake:
+    # Check if message is in database, if it is change the was_seen status to seen
     if message := ContactFormMessage.query.filter_by(id=message_id).first():
-        message.was_seen = True if message.was_seen is False else False
+        message.was_seen = True
```
Comment (jacobduncan00) [inline api/admin.py:225]: Sounds good, assuming we are handling the case when a message has been seen and still gets clicked on the frontend which I think is a valid assumption

**Comment by jacobduncan00** [comment]
✅

**Review by jventura1738 — APPROVED** [review]
API changes all good.  @jacobduncan00 approve for the front-end when possible?

**Pull request merged**